### PR TITLE
Move submodules to relative paths

### PR DESCRIPTION
**Pull request by pljones**

### CODE REVIEW

**Review by ann0see — APPROVED** [review]
Looks ok to me.

File: .gitmodules [hunk 1]
```diff
@@ -1,24 +1,24 @@
 [submodule "libs/oboe"]
 	path = libs/oboe
-	url = https://github.com/google/oboe.git
+	url = git@github.com:google/oboe.git
```
Comment (hoffie) [inline .gitmodules:3]:
```suggestion
	url = ../../google/oboe.git
```

File: .gitmodules [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
 [submodule "tools/jamulus-php"]
 	path = tools/jamulus-php
-	url = https://github.com/softins/jamulus-php
+	url = git@github.com:softins/jamulus-php.git
```
Comment (hoffie) [inline .gitmodules:6]:
```suggestion
	url = ../../softins/jamulus-php.git
```

File: .gitmodules [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
 [submodule "tools/jamulus-web"]
 	path = tools/jamulus-web
-	url = https://github.com/softins/jamulus-web
+	url = git@github.com:softins/jamulus-web.git
```
Comment (hoffie) [inline .gitmodules:9]:
```suggestion
	url = ../../softins/jamulus-web.git
```

File: .gitmodules [hunk 4]
```diff
@@ -1,24 +1,24 @@
...
 [submodule "tools/jamulus-wireshark"]
 	path = tools/jamulus-wireshark
-	url = https://github.com/softins/jamulus-wireshark
+	url = git@github.com:softins/jamulus-wireshark.git
```
Comment (hoffie) [inline .gitmodules:12]:
```suggestion
	url = ../../softins/jamulus-wireshark.git
```

File: .gitmodules [hunk 5]
```diff
@@ -1,24 +1,24 @@
...
-	url = https://github.com/pljones/jamulus-historytool
+	url = git@github.com:pljones/jamulus-historytool.git
 [submodule "tools/jamulus-jamexporter"]
 	path = tools/jamulus-jamexporter
-	url = https://github.com/pljones/jamulus-jamexporter
+	url = git@github.com:pljones/jamulus-jamexporter.git
 [submodule "tools/jamulus-docker"]
 	path = tools/jamulus-docker
-	url = https://github.com/grundic/jamulus-docker
+	url = git@github.com:grundic/jamulus-docker.git
```
Comment (hoffie) [inline .gitmodules:21]:
```suggestion
	url = ../../grundic/jamulus-docker.git
```

File: .gitmodules [hunk 6]
```diff
@@ -1,24 +1,24 @@
...
 [submodule "tools/jamulus-server-remote"]
 	path = tools/jamulus-server-remote
-	url = https://github.com/vdellamea/jamulus-server-remote
+	url = git@github.com:vdellamea/jamulus-server-remote.git
```
Comment (hoffie) [inline .gitmodules:24]:
```suggestion
	url = ../../vdellamea/jamulus-server-remote.git
```

File: .gitmodules [hunk 7]
```diff
@@ -1,24 +1,24 @@
...
-	url = https://github.com/softins/jamulus-wireshark
+	url = git@github.com:softins/jamulus-wireshark.git
 [submodule "tools/jamulus-historytool"]
 	path = tools/jamulus-historytool
-	url = https://github.com/pljones/jamulus-historytool
+	url = git@github.com:pljones/jamulus-historytool.git
 [submodule "tools/jamulus-jamexporter"]
 	path = tools/jamulus-jamexporter
-	url = https://github.com/pljones/jamulus-jamexporter
+	url = git@github.com:pljones/jamulus-jamexporter.git
```
Comment (hoffie) [inline .gitmodules:18]:
```suggestion
	url = ../../pljones/jamulus-jamexporter.git
```

File: .gitmodules [hunk 8]
```diff
@@ -1,24 +1,24 @@
...
-	url = https://github.com/softins/jamulus-web
+	url = git@github.com:softins/jamulus-web.git
 [submodule "tools/jamulus-wireshark"]
 	path = tools/jamulus-wireshark
-	url = https://github.com/softins/jamulus-wireshark
+	url = git@github.com:softins/jamulus-wireshark.git
 [submodule "tools/jamulus-historytool"]
 	path = tools/jamulus-historytool
-	url = https://github.com/pljones/jamulus-historytool
+	url = git@github.com:pljones/jamulus-historytool.git
```
Comment (hoffie) [inline .gitmodules:15]:
```suggestion
	url = ../../pljones/jamulus-historytool.git
```

**Review by hoffie — COMMENTED** [review]
Not sure if relevant, but this could make it a bit harder to use for anonymous Github users (i.e. Jamulus repo cloned via http):

Before this PR:
```
$ git clone https://github.com/jamulussoftware/jamulus.git && cd jamulus
[...]
$ git submodule init && git submodule update
Submodule 'libs/oboe' (https://github.com/google/oboe.git) registered for path 'libs/oboe'
[...]
Cloning into '/tmp/jamulus/libs/oboe'...
[...]
Submodule path 'libs/oboe': checked out '6c6bff4c5a67b5da3067dc8b1480c2e310d81a2e'
[...]
```

After this PR:
```
$ git clone https://github.com/jamulussoftware/jamulus.git && cd jamulus
[...]

$ git fetch origin refs/pull/934/head:pr934 && git checkout pr934
[...]
Switched to branch 'pr934'

$ git submodule init && git submodule update
Submodule 'libs/oboe' (git@github.com:google/oboe.git) registered for path 'libs/oboe'
[...]
Cloning into '/tmp/jamulus/libs/oboe'...
Enter passphrase for key '...:  # would break if I hadn't set up ssh for github.com
```

[Someone StackOverflow](https://stackoverflow.com/questions/40841882/automatically-access-git-submodules-via-ssh-or-https/44630028#44630028) suggests using relative paths (which seem to be considered relative to origin, not the file system) and this seems to work in a quick test.

**Comment by pljones** [comment]
Github are removing https clone access "soon".  I expect it's so they can ensure they collect everyone's email address - i.e. you have to register to download source.

**Comment by pljones** [comment]
OK, had time to check this out and moved to relative paths as suggested.  Thanks!

**Pull request merged**